server/redis.repository.js

```diff
@@ -0,0 +1,79 @@
+const redis = require("redis");
+const { promisify } = require("util");
+const client = redis.createClient();
+const hgetAsync = promisify(client.hget).bind(client);
+const hincrAsync = promisify(client.hincrby).bind(client);
+const hsetAsync = promisify(client.hset).bind(client);
+
+/**
+ * Creates a room key in Redis and sets:
+ *      1. the number of movies
+ *      2. the number of guests
+ *      3. the number of total swipes to 0
+ * 
+ * @param {String} roomId - The room ID.
+ * @param {Integer} numGuests - The number of guests in the room, including the host
+ * @param {Object[]} movies - The list of movies with the respective movie Ids
+ * @returns {Integer} x - Number of fields sets (No errors if greater than 0)
+ */
+const initializeRoom = async (roomId, numGuests, movies) => {
+    let roomArgs = ["numUsers", numGuests, "total_swipes", 0, "numMovies", movies.length];
+    movies.forEach(element => {
```

Comment:
why do we push each movie id followed by a 0? Is this a key value pair where the value is the amount of swipes it has and the key is the movie id?

Reply:
I was also quite confused by this, since the node redis library docs only mentions hmset, but [here in the redis docs ](https://redis.io/commands/hmset), we see that HMSET is actually deprecated and HSET should be used instead.  Also, the [docs for this library](https://www.npmjs.com/package/redis#commands) say that the methods of the client object are 1:1 mapping to the actual redis commands (on command line)

In the [redis docs](https://redis.io/commands/hset) it says that the array of args (here we have roomArgs) sent in to HGET would be interpreted as key value sequentially, so like what Bill has here. 

@billxsheng  Does this sound about right?

Reply:
Exactly! I pushed 0 into the array after each movie to initialize it the value to 0 (since all movies start with 0 likes after the session is begun). The mapping with the HGET function are sequential as @dlhxu has stated. Therefore, pushing an array as `[movie_1, 0, movie_2, 0]` is essentially `{movie_1: 0, movie_2: 0}` within Redis. @jeffreyquiambao let me know if that makes sense!